collections/_tutorials/2022-09-12-tutorial-cvae.md

```diff
@@ -0,0 +1,296 @@
+The following observations can be made:
+* since $$0\leq p(y \vert x) \leq 1$$, $$log\left(p(y \vert x)\right) \leq 0$$
+
+* since $$x$$ and $$y$$ are known, $$log\left(p(y \vert x)\right)$$ is a fixed value
+
+* by definition $$D_{KL}\left(q(z \vert x,y) \parallel p(z \vert x,y) \right) \geq 0$$
+
+* since $$\mathcal{L} = -D_{KL}\left(q(z \vert x,y) \parallel p(y,z \vert x)\right)$$, $$\mathcal{L} \leq 0$$
+
+
+The previous expression can thus be rewritten as follows:
+
+$$\underbrace{D_{KL}\left(q(z \vert x,y) \parallel p(z \vert x,y) \right)}_{\geq 0} \,+\, \underbrace{\mathcal{L}}_{\leq 0} \,=\, \underbrace{log\left(p(y \vert x)\right)}_{\leq 0 \,\, \text{and fixed}}$$
+
+>Thus, by tweaking $$q(z \vert x,y)$$, we can seek to maximize the ELBO $$\mathcal{L}$$, which will imply the minimization of the KL divergence $$D_{KL}\left(q(z \vert x,y) \parallel p(z \vert x,y) \right)$$, and consequently a distribution $$q(z \vert x,y)$$ that is close to $$p(z \vert x,y)$$.
+
+&nbsp;
+
+### ELBO reformulation
+
+The ELBO $$\mathcal{L}$$ must be reformulated to justify the loss involved in the conditional VAE framework. The corresponding derivation is provided below.
+
+$$\mathcal{L} = \int{q(z \vert x,y) \cdot log\left(\frac{p(y,z \vert x)}{q(z \vert x,y)}\right) \,dz}$$
+
+By using the following ***conditional probability*** relations:
+
+$$
+p(x,y,z) = \left\{
+  \begin{array}
+    pp(y,z \vert x) \cdot p(x) \\
+    p(y \vert x,z) \cdot p(x,z)
+  \end{array}
+  \right.
+$$
+
+$$p(x,y) = p(z \vert x) \cdot p(x)$$
+
+the next equations can be easily obtained
+
+$$p(y,z \vert x) = \frac{p(y \vert x,z) \cdot p(x,z)}{p(x)}$$
+
+$$p(y,z \vert x) = \frac{p(y \vert x,z) \cdot p(z \vert x) \cdot p(x)}{p(x)}$$
+
+$$p(y,z \vert x) = p(y \vert x,z) \cdot p(z \vert x)$$
+
+&nbsp;
+
+The ELBO $$\mathcal{L}$$ expression can thus be rewritten as
+
+$$\mathcal{L} = \int{q(z \vert x,y) \cdot log\left(\frac{p(y \vert x,z) \cdot p(z \vert x)}{q(z \vert x,y)}\right) \,dz}$$
+
+$$\mathcal{L} = \int{q(z \vert x,y) \cdot \left[log\left(p(y \vert x,z)\right) + log\left(\frac{p(z \vert x)}{q(z \vert x,y)}\right) \right] \,dz}$$
+
+$$\mathcal{L} = \int{q(z \vert x,y) \cdot log\left(p(y \vert x,z)\right) \,dz} \,+\, \int{q(z \vert x,y) \cdot log\left(\frac{p(z \vert x)}{q(z \vert x,y)}\right) \,dz}$$
+
+$$\mathcal{L} =  \mathbb{E}_{z\sim q(z \vert x,y)} \left[log\left(p(y \vert x,z)\right)\right] - D_{KL}\left(q(z \vert x,y)\parallel p(z \vert x)\right)$$
+
+where $$\mathbb{E}_{z\sim q(z \vert x,y)}$$ is the mathematical expectation with respect to $$q(z \vert x,y)$$. 
+
+&nbsp;
+
+At this stage of analysis, it is necessary to model $$p(y \vert x,y)$$ via an ***a priori distribution***. Let's first note that $$p(y \vert x,z)$$ is modeled by a neural network $$f(\cdot)$$ so that $$\hat{y}=f(x,z)$$. Since this function is deterministic, it will allow to model $$p\left(y \vert \hat{y}\right)$$. By approximating $$p\left(y \vert \hat{y}\right)$$ by a Bernoulli distribution, we have
```

Comment:
p(y|x,y) => p(y|x,z)

Reply:
on pourrait en parler, mais l'affirmation "that p(y|x,z) is modeled by a neural network f(⋅) so that y^=f(x,z)" est fausse (ou bizare à tout le moins) à mon avis.  Si p c'est une probabilité et y est une variable, alors y^ est une variable ou une probabilité?